src/main/scala/Day4/CreditCard.scala

```diff
@@ -0,0 +1,34 @@
+package Day4
+
+import scala.collection.mutable.ListBuffer
+
+object CreditCard extends App{
+
+  def addCheckDigit(identifyingNo: String, checkDigit: Int) = {
+    (identifyingNo + checkDigit).map(_.asDigit).toList.reverse.to[ListBuffer]
+  }
+
+  def validateIdentifyingNumber(numberWithCheck: ListBuffer[Int], index:Int = 1):Boolean = {
+    if (index <= (numberWithCheck.length-1) ) {
+      if (index % 2 != 0) {
+        numberWithCheck.update(index, numberWithCheck(index)*2)
+        validateIdentifyingNumber(numberWithCheck: ListBuffer[Int], index + 1)
+      } else {
+        validateIdentifyingNumber(numberWithCheck: ListBuffer[Int], index + 1)
```

Comment:
`validateIdentifyingNumber(numberWithCheck: ListBuffer[Int], index + 1)` you're doing this both in the inner if and else, wouldn't it make sense to just do it outside of the if and once as you're doing it either way any way?